src/librustdoc/html/static/main.js

```diff
@@ -1970,18 +1970,17 @@ function defocusSearchBar() {
      *   * Replace "<code>foo</code>" with "`code`"
      *   * Strip all other HTML tags
      *
-     * Note that it just uses a regex to strip the other HTML tags,
-     * so it may not work correctly for cases where there is an angle
-     * bracket inside quotes.
-     *
      * Used by the dynamic sidebar crate list renderer.
      *
      * @param  {[string]} html [The HTML to convert]
      * @return {[string]}      [The resulting plaintext]
      */
     function convertHTMLToPlaintext(html) {
-        return html.replace('<code>', '`').replace('</code>', '`')
-                   .replace(/<[^>]+>/, '');
+        var dom = new DOMParser().parseFromString(
+            html.replace('<code>', '`').replace('</code>', '`'),
```

Comment:
This bit is no longer necessary I think.
```suggestion
            html,
```

Reply:
Oh I take it back - it is necessary if you want to replace code blocks with \`. Not sure why rustdoc is doing that, though.

Reply:
It's not strictly necessary, but I think it's better to do it this way, for one since that's how it used to work.

Reply:
Well, it *kinda* makes sense. However, we also keep other HTML tags like `<b>` and `<i>`. Should we also "revert" them into markdown too?

Reply:
> It's not strictly necessary, but I think it's better to do it this way, for one since it used to use ` here.

My comment still stands. :p

Reply:
I really don't think it makes sense to parse HTML ourselves - we have the same issue as in https://github.com/rust-lang/rust/pull/68699#issuecomment-627465641 that we don't know what was originally html and what was markdown. If we don't want to show rendered text, we shouldn't show it, but I don't think we should try to rewrite some sort of HTML to markdown converter.

Reply:
> we have the same issue as in [#68699 (comment)](https://github.com/rust-lang/rust/pull/68699#issuecomment-627465641) that we don't know what was originally html and what was markdown

I do not think that is correct. The example given in https://github.com/rust-lang/rust/pull/68699#issuecomment-627465641 was:

```rust
/// foo `this is a code span` bar \`this isn't\` baz
pub struct Foo;
```

That is stored in the search index as:

```html
foo <code>this is a code span</code> bar `this isn't` baz
```

And then `convertHTMLToPlaintext` converts that into:

```
foo `this is a code span` bar `this isn't` baz
```

Which is *exactly* the same as the current hover text.

-----

That said, I am open to just stripping every HTML tag if we decide that would be better. I just don't think https://github.com/rust-lang/rust/pull/68699#issuecomment-627465641 is relevant.

Reply:
@camelid the test case is the other way:

```rust
/// foo <code>this is a code span</code> in HTML
///
/// `this is markdown` baz
pub struct Foo;
```

There's no way to get back that info after the fact.

Reply:
I just copy-pasted the test case from the comment you linked to. The comment doesn't seem to be taking issue with the example you're giving. This is the full comment text:

> No. The summaries stored in the search index don't contain that information. For:
>
> ```rust
> /// foo `this is a code span` bar \`this isn't\` baz
> pub struct Foo;
> ```
>
> the search index contains: 
>
> ```text
> foo `this is a code span` bar `this isn't` baz
> ```
>
> There is no way in JavaScript to know which backticks meant codeblocks and which didn't.
>
> As I've said, to do this right the summaries need to be rendered and stored in the search index as HTML.

It is saying that converting backticks into HTML is a problem because they both mean inline code and literal backticks. Converting in the opposite direction is loss-less.

-----

And, either way, I don't see how converting both `` `code` `` and `<code>code</code>` into `` `code` `` when this is just for alt-text is an issue. If for some reason this is a blocker, I am absolutely fine to just strip all HTML tags.

Reply:
I'm not 100% sure about this but I don't think it should block this change. So if @GuillaumeGomez thinks this is good to go I do too.

Reply:
I said I didn't. :stuck_out_tongue: